transports/noise/src/lib.rs

```diff
@@ -88,6 +89,15 @@ impl<H, C: Zeroize, R> NoiseConfig<H, C, R> {
         NoiseAuthenticated { config: self }
     }
 
+    /// Set the noise prologue.
+    ///
+    /// The prologue can contain arbitrary data and will be hashed into the noise handshake.
+    /// For the handshake to succeed, both parties must set the same prologue.
+    pub fn with_prologue(&mut self, prologue: Vec<u8>) -> &mut Self {
+        self.prologue = prologue;
+        self
+    }
```

Comment:
I made this a `with_` function instead of exposing it in every constructor because it is often not needed and this way, we can keep the entire prologue functionality hidden from users that just want to use regular noise for their connections.

Having said that, if we agree to merge https://github.com/libp2p/rust-libp2p/pull/2887, then most users don't need to touch `NoiseConfig` anyway so perhaps it should just be a regular constructor parameter?